### PR TITLE
fix(ios): prevents installation of packages without JS

### LINKED ISSUE #5694: bug(ios): Desktop-only keyboard package does not raise issue during install
Sentry Issue: [KEYMAN-IOS-16](https://sentry.keyman.com/organizations/keyman/issues/11444/?referrer=github_integration)

```
File missing for keyboard
```

This is arising for the [Hieroglyphic package](https://keyman.com/keyboards/hieroglyphic), which does not contain a JS keyboard.  Thus, we can infer that the package is "successfully" installing, with the iOS KeymanEngine's installer failing to note that the package doesn't actually contain a usable keyboard.

It's a "silent failure" until the user tries to actually _use_ the Hieroglyphic keyboard.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #5694.

Also fixes up how package-opening errors are handled and forwarded.  The package parser wasn't actually surfacing errors that arose internally, causing any error condition to report a "corrupted package," even when the cause was different.

Adds two new strings that'll need localization - these provide better, more specific messages for package problems:
- A "package is missing" message.  Shouldn't ever be seen from the app, but then again, most errors "shouldn't ever be seen".  It's best to keep it around "just in case."
- A "package is only for desktop" message.  (See `TEST_HIEROGLYPHIC` below.)

## User Testing

TEST_HIEROGLYPHIC:  Attempt to install the hieroglyphic package from https://keyman.com/keyboards/hieroglyphic.

You should see a popup message like the following:

image

But with this text instead:

> "This package does not include support for your device."

TEST_CAM_QWERTY:  Attempt to install the `sil_cameroon_qwerty` package from https://keyman.com/keyboards/sil_cameroon_qwerty.

It should install successfully.

### CODE REVIEW

File: ios/engine/KMEI/KeymanEngine/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -118,12 +118,18 @@
...
+"kmp-error-missing" = "The specified package does not exist.";
+
 /* Error installing a Keyman package - expected resource (keyboard or dictionary) is missing */
 "kmp-error-missing-resource" = "This package does not contain the requested keyboard or dictionary.";
 
 /* Error opening a Keyman package - cannot parse contents */
 "kmp-error-no-metadata" = "This package was not properly built - contents unknown.";
 
+/* Error opening a Keyman package - package's contents are for desktop platforms only */
+"kmp-error-unsupported" = "The package's contents are not supported on mobile devices.";
```
Comment (mcdurdin) [inline ios/engine/KMEI/KeymanEngine/en.lproj/Localizable.strings:131]:
```suggestion
"kmp-error-unsupported" = "This package does not include support for your device.";
```

**Review by mcdurdin — COMMENTED** [review]
LGTM pending user testing.

Reworded one error message to try and make it a little clearer.

**Review by mcdurdin — APPROVED** [review]

**Comment by MakaraSok** [comment]
* **TEST_HIEROGLYPHIC: FAILED** OK when double click on .kmp, but not when "Install from File".
    * The error would show up only when one attempts to install the keyboard by double-clicking on the downloaded `hieroglyphic.kmp`. When trying and installing the .kmp from Settings > Install from File, no error is shown. The keyboard was installed and no notification.
    * The message shown is like that of what's on the screenshot, not "This package does not include support for your device." as indicated -- because the build hasn't got through yet.
      image
* **TEST_CAM_QWERTY: PASSED** The keyboard got installed successfully either way.

**Comment by MakaraSok** [comment]
* **TEST_HIEROGLYPHIC** both ways of installing would give the same expected error when a keyboard is not support on iOS.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.114-alpha](https://keyman.com/downloads/releases/alpha/15.0.114)

**Pull request merged**